### PR TITLE
Eslint automation fixes

### LINKED ISSUE #216: Replace deprecated eslint GitHub action with non-deprecated one
While[ troubleshooting a lodash error in some recent runs of the eslint GitHub action](https://github.com/ourjapanlife/findadoc-frontend/pull/203/checks?check_run_id=2977418423), I noticed that the action we refer to is deprecated: https://github.com/hallee/eslint-action

> This repository has been abandoned
GitHub now offers a much cleaner way to handle linters with complex projects and provide annotations on PRs. The problem matcher format is now the best way to handle this.
> 
> actions/setup-node provides ESLint annotations by default now, and it is the best solution for lint annotations in PRs. See https://github.com/actions/setup-node/blob/main/.github/eslint-stylish.json as an example of the problem matcher format.

Let's use the problem matcher format! I have turned off the eslint action to prevent failing builds

Update: the problem matcher docs are confusing. Opting for simplicity

### PR DESCRIPTION
**Pull request by ann-kilzer**
## Resolves #216 

- [x] Try the [Problem Matcher format](https://github.com/actions/toolkit/blob/master/docs/problem-matchers.md)
- [x] remove deprecated action in favor of simpler approach

### How to test

Check the actions on this PR

### CODE REVIEW

File: components/ContributorsList.vue [hunk 1]
```diff
@@ -19,6 +19,8 @@
 <script>
 import { leads } from "../data/contributors.json";
 
+var j = 0;
```
Comment (ann-kilzer) [inline components/ContributorsList.vue:22]: The linter is catching these: https://github.com/ourjapanlife/findadoc-frontend/pull/217/checks?check_run_id=2978359608

File: .github/workflows/eslint.yml [hunk 2]
```diff
@@ -7,10 +7,9 @@ jobs:
...
-        # GITHUB_TOKEN in forked repositories is read-only
-        # https://help.github.com/en/actions/reference/events-that-trigger-workflows#pull-request-event-pull_request
-        if: ${{ github.event_name == 'push' || github.event.pull_request.head.repo.full_name == github.repository }} 
-        with:
-          repo-token: ${{secrets.GITHUB_TOKEN}}
-          source-root: web
+      - name: Install modules
+        run: yarn
+      - name: Register Matcher
+        run: echo "::add-matcher::.github/eslint-compact.json"
```
Comment (ermish) [inline .github/workflows/eslint.yml:13]: I've never seen this before! What is the compact file doing?

**Review by ermish — CHANGES_REQUESTED** [review]
Not exactly sure what it's doing, so it might be good to add a few explanatory comments

File: .github/workflows/eslint.yml (hunk 2 above)
Comment (ann-kilzer) [reply .github/workflows/eslint.yml:13]: It's supposed to add annotations. I am having trouble getting it working and might simplify.

https://github.com/actions/toolkit/blob/main/docs/commands.md#problem-matchers

**Review by ermish — APPROVED** [review]
LGTM! nice and intuitive!

**Pull request merged**